### PR TITLE
I18n setup

### PR DESCRIPTION
**Pull request by yagotome**
- Setting up [i18next](https://www.i18next.com/) for multi-lingual strings
- Create pattern for string resources (a namespace for each page and an auxiliary common namespace)
- Add some usage examples within the pages

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-/**
- * Implement Gatsby's Node APIs in this file.
- *
- * See: https://www.gatsbyjs.org/docs/node-apis/
- */
+const fs = require('fs-extra');
+const path = require('path');
```
Comment (BelfDev) [inline gatsby-node.js]: Are we going to prioritize imports via the "require" method or are we going to adopt the full ES6 [importing style](https://hackernoon.com/import-export-default-require-commandjs-javascript-nodejs-es6-vs-cheatsheet-different-tutorial-example-5a321738b50f)?

File: package.json [hunk 2]
```diff
@@ -15,11 +15,15 @@
     "gatsby-plugin-sharp": "^2.4.3",
     "gatsby-source-filesystem": "^2.1.46",
     "gatsby-transformer-sharp": "^2.3.13",
+    "i18next": "^19.2.0",
```
Comment (BelfDev) [inline package.json]: Did you consider Gatsby `i18n` plugins or `react-intl`?
https://www.gatsbyjs.org/docs/localization-i18n/

File: src/pages/index.js [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
+  const { t } = useTranslation('IndexPage');
+
+  return (
+    <Layout>
+      <SEO title="Home" />
+      <h1>{t('greetings')}</h1>
+      <div style={{ maxWidth: '300px', marginBottom: '1.45rem' }}>
+        <Image />
+      </div>
+      <Link to="/page-2/">{t('common:go_to_page_2')}</Link>
```
Comment (BelfDev) [inline src/pages/index.js]: Goshhhh -- looks damn ugly XD

File: gatsby-node.js (hunk 1 above)
Comment (yagotome) [reply gatsby-node.js]: ES6 Modules for browser code. It turns out that this file runs at build time only in a node.js environment, which doesn't support ES6 Modules.

File: package.json (hunk 2 above)
Comment (yagotome) [reply package.json]: Yeah! However their design requires us to keep a page file for each language we have. This duplication could cause us some headaches in our maintenance.

Comment (BelfDev) [reply package.json]: Totally agree!

**Pull request merged**